references/cuegen/util_test.go

```diff
@@ -0,0 +1,128 @@
+		{name: "Test", isDef: true, want: "#Test"},
+		{name: "Test", isDef: false, want: "Test"},
+	}
+
+	for _, tt := range tests {
+		assert.Equal(t, tt.want, ident(tt.name, tt.isDef).String())
+	}
+}
+
+func TestBasicType(t *testing.T) {
+	tests := []struct {
+		name string
+		typ  *gotypes.Basic
+		want string
+	}{
+		{name: "uintptr", typ: gotypes.Typ[gotypes.Uintptr], want: "uint64"},
+		{name: "byte", typ: gotypes.Typ[gotypes.Byte], want: "uint8"},
+		{name: "int", typ: gotypes.Typ[gotypes.Int], want: "int"},
+		{name: "int8", typ: gotypes.Typ[gotypes.Int8], want: "int8"},
+		{name: "int16", typ: gotypes.Typ[gotypes.Int16], want: "int16"},
+		{name: "int32", typ: gotypes.Typ[gotypes.Int32], want: "int32"},
+		{name: "int64", typ: gotypes.Typ[gotypes.Int64], want: "int64"},
+		{name: "uint", typ: gotypes.Typ[gotypes.Uint], want: "uint"},
+		{name: "uint8", typ: gotypes.Typ[gotypes.Uint8], want: "uint8"},
+		{name: "uint16", typ: gotypes.Typ[gotypes.Uint16], want: "uint16"},
+		{name: "uint32", typ: gotypes.Typ[gotypes.Uint32], want: "uint32"},
+		{name: "uint64", typ: gotypes.Typ[gotypes.Uint64], want: "uint64"},
+		{name: "float32", typ: gotypes.Typ[gotypes.Float32], want: "float32"},
+		{name: "float64", typ: gotypes.Typ[gotypes.Float64], want: "float64"},
+		{name: "string", typ: gotypes.Typ[gotypes.String], want: "string"},
+		{name: "bool", typ: gotypes.Typ[gotypes.Bool], want: "bool"},
+	}
+
+	for _, tt := range tests {
+		assert.Equal(t, tt.want, basicType(tt.typ).(*cueast.Ident).String(), tt.name)
+	}
+}
+
+func TestAnyLit(t *testing.T) {
+	assert.Equal(t, anyLit(), &cueast.StructLit{Elts: []cueast.Decl{&cueast.Ellipsis{}}})
+}
+
+func TestBasicLabel(t *testing.T) {
+	overflowInt64 := strconv.FormatInt(math.MaxInt64, 10) + "0"
+	overflowUint64 := strconv.FormatUint(math.MaxUint64, 10) + "0"
+	overflowFloat64 := strconv.FormatFloat(math.MaxFloat64, 'f', -1, 64) + "0"
+
+	tests := []struct {
+		name    string
+		typ     *gotypes.Basic
+		v       string
+		wantErr bool
+		want    *cueast.BasicLit
+	}{
+		{name: "int", typ: gotypes.Typ[gotypes.Int], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "int8", typ: gotypes.Typ[gotypes.Int8], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "int16", typ: gotypes.Typ[gotypes.Int16], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "int32", typ: gotypes.Typ[gotypes.Int32], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "int64", typ: gotypes.Typ[gotypes.Int64], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "uint", typ: gotypes.Typ[gotypes.Uint], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
+		{name: "uint8", typ: gotypes.Typ[gotypes.Uint8], v: "123", wantErr: false, want: &cueast.BasicLit{Kind: cuetoken.INT, Value: "123"}},
```

Comment:
Is this the byte case?

Reply:
In the unit test I used `gotypes.Typ` to quickly generate `*gotypes.Basic`. However, in gotypes, `Byte = Uint8` https://github.com/golang/go/blob/master/src/go/types/basic.go#L45. 

So `gotypes.Typ[gotypes.Byte].String() == "uint8"`, it doesn't go through `case "byte` at all. Only when goast parses the file, it appears `Type.String() == "byte"` .

Reply:
So in this unit test, I did not test `byte` case.